public/src/components/teacher/Results.tsx

```diff
@@ -171,18 +171,20 @@ export class Results extends React.Component<IResultsProps, IResultsState> {
     }
 
     private async rebuildSubmission(): Promise<boolean> {
-        const currentSubmission = this.state.selectedSubmission;
+        const currentSubmission = this.state.selectedSubmission
         if (currentSubmission && currentSubmission.submission) {
-            const ans = await this.props.onSubmissionRebuild(currentSubmission.assignment.getId(), currentSubmission.submission.id);
+            const ans = await this.props.onSubmissionRebuild(currentSubmission.assignment.getId(), currentSubmission.submission.id)
             if (ans) {
-                currentSubmission.submission = ans;
+                // TODO: This no longer works because the updated submission
```

Comment:
Couldn't this be fixed by simply fetching the submission using `GetSubmission`? That would at least keep the functionality as it was.